plugins/org.polarsys.capella.vp.requirements.ui.commands/src/org/polarsys/capella/vp/requirements/ui/commands/messages.properties

```diff
@@ -2,8 +2,8 @@ GROUP_TITLE_FOR_LABEL_AREA=Requirement's label
 GROUP_TITLE_FOR_CONTENT_AREA=Requirement's content
 EXPRESSION_LABEL_TEXT=Expression
 LENGTH_LABEL_TEXT=Length (put nothing to display full text):
-DefaultValueOfLabelExpression=aql:self.ownedAttributes->select( a | a.definition.ReqIFLongName == 'IE PUID').value
-DefaultValueOfContentExpression=aql:OrderedSet{self.ReqIFLongName, self.ReqIFName, self.ReqIFChapterName, self.ReqIFText}->select(s  | s != 'null' and s.size() > 0).prefix('- ')->sep('\\n')
+DefaultValueOfLabelExpression=aql:self.getDefaultRequirementTitle()

```

Comment:
note that the user wants to customize its expressions. it is possible to call an aql expression for the text displayed in the Project Explorer, and on diagrams, title and content of requirements.
https://github.com/eclipse-capella/capella-requirements-vp/blob/master/plugins/org.polarsys.capella.vp.requirements.doc/html/3_userguide/34_preferences.mediawiki#requirement-label
and 
https://github.com/eclipse-capella/capella-requirements-vp/blob/master/plugins/org.polarsys.capella.vp.requirements.doc/html/3_userguide/311_diagrams.mediawiki#label-and-content-configuration


if we use a service like proposed, then it can't customize it easily except by looking at source code on how is it coded.

one possible solution might be to do provide him services like below to ease the customisation.
title:
aql:self.toOneLinexxx("IE PUID", "ReqIfLongName")

and something like for content:
aql:self.separatexxx("reqIFLongName", "reqIFName", "reqIFChapterName", "reqIFText")

note that those services shall work also on Project Explorer expression. (called on https://github.com/eclipse-capella/capella-requirements-vp/blob/master/plugins/org.polarsys.kitalpha.vp.requirements.model.edit.decorators/src/org/polarsys/kitalpha/vp/requirements/Requirements/provider/RequirementItemProviderDecorator.java#L73 )


Dont forget to update the documentation to explain how to call those services.

Note that current models that will migrate to 7.0 **must not** have to update manually their expressions, 
so if you just do a reduceString in CapellaRequirementsOpenJavaService.evaluateExpression 
the current models will lose the call to the transformHTML stuff and their display will be broken.
you need to contribute to the migration to migrate the expressions (that may have changed from the default expression)
(to call the removed transformHTML method or to call the new services)

Reply:
Indeed, you raise a point we hadn't thought of with the expression update and migration ;)